pkg/producer/convertor.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package producer
+
+import (
+	"net"
+
+	"k8s.io/klog/v2"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/producer/protobuf"
+)
+
+// ConvertIPFIXToKafka is an interface to support multiple proto schema for Kafka
+// producer.
+type ConvertIPFIXToKafka interface {
+	// convertIPFIXMsgToFlowMsgs converts multiple data records in the IPFIX message
+	// to flow messages for any given proto schema.
+	convertIPFIXMsgToFlowMsgs(msg *entities.Message) []*protobuf.FlowMessage
+}
+
+func NewConvertorForProtoSchema(schemaType string) ConvertIPFIXToKafka {
+	switch schemaType {
+	case "FlowType1":
```

Comment:
I feel cleaner to define a map and add schemaType -> converter mapping there, or you can define a registration func for each schema to register itself.

Reply:
Addressed. Please take a look.